image/tarexport/save.go

```diff
@@ -23,14 +23,16 @@ import (
 type imageDescriptor struct {
 	refs   []reference.NamedTagged
 	layers []string
+	image  *image.Image
```

Comment:
Keep the layer reference also in this object. Either leave both image and layers empty in `parseNamed` and fill them in a loop before `save` or if you fill them in `parseNamed` make sure they are released if `parseNamed` returns an error. Should avoid that one method returns image IDs that are already invalid for getting the layer refs by the time next method runs.